### PR TITLE
Wait for Elasticsearch until a connection can be made on startup with version probe enabled

### PR DESCRIPTION
**Pull request by janheise**
Added documentation fort the new configuration options for Elasticsearch to configure the behavior with version probing enabled. Also added the changed behavior the the Upgrade doc

### CODE REVIEW

File: pages/upgrade/graylog-4.1.rst [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+   :backlinks: top
+
+
+Breaking Changes
+================
+
+Changes to the Elasticsearch Support
+------------------------------------
+
+When you have version-probing for the used Elasticsearch version enabled, and Graylog starts up but can not connect to ES, the startup stopped immediately with v4.0 and prior. Starting from 4.1 the default behaviour is, that Graylog retries connecting with a delay until it can connect to Elasticsearch. See the Elasticsearch configuration page for details.
```
Comment (dennisoelkers) [inline pages/upgrade/graylog-4.1.rst]: This is slightly inaccurate. Before v4.0, there was no version probing, but prior to v4.0 _and_ in v4.0 _without_ version probing, Graylog continues connecting to ES until it is successfull, so this change is a breaking change compared to v4.0, but makes the behavior consistent with versions v3.3 and before _and_ with the behavior when version probing is disabled.

**Review by dennisoelkers — CHANGES_REQUESTED** [review]
Looks good, one minor remark.

**Comment by bernd** [comment]
@janheise Can you resolve the conflicts and address @dennisoelkers comment so we can merge this for 4.1? Thank you! :slightly_smiling_face:

**Review by dennisoelkers — APPROVED** [review]
Looks good! :tada:

**Pull request merged**